spec/system/passenger_registration_spec.rb

```diff
@@ -2,64 +2,88 @@
-        expect { submit }.to(change { Passenger.count }.by(1))
-        expect(page).to have_text 'Passenger registration successful'
+      end
+
+      it 'creates a passenger' do
+        expect { submit }.to change(Passenger, :count).by(1)
+      end
+
+      it 'creates a pending passenger' do
+        submit
         expect(Passenger.last).to be_pending
       end
+
+      it 'informs you of success' do
+        submit
+        expect(page).to have_text 'Passenger registration successful'
+      end
     end
 
-    context 'unsuccessfully' do
-      it 'renders errors in the flash' do
-        visit register_passengers_path
+    context 'with errors' do
+      before do
         fill_in 'Address', with: '123 turkey lane'
-        expect { submit }.not_to(change { Passenger.count })
+      end
+
+      it 'does not create a new passenger' do
+        expect { submit }.not_to change(Passenger, :count)
+      end
+
+      it 'renders errors in the flash' do
+        submit
         expect(page).to have_text "Phone Number can't be blank"
       end
     end
   end
 
-  context 'editing registration' do
-    context 'while still pending' do
-      it 'redirects to the edit page and allows editing' do
-        @passenger = create(:passenger, registration_status: 'pending')
-        login_as(@passenger)
-        visit edit_passenger_path(@passenger)
+  context 'when editing a registration' do
+    before do
+      login_as passenger
+      visit edit_passenger_path(passenger)
+    end
+
+    context 'when still pending' do
+      let(:passenger) { create(:passenger, registration_status: 'pending') }
+
+      it 'allows editing' do
         expect(page).to have_field 'Address'
-        expect(page).to have_current_path edit_passenger_path(@passenger)
+      end
+
+      it 'redirects to the edit page' do
         visit register_passengers_path
-        expect(page).to have_field 'Address'
-        expect(page).to have_current_path edit_passenger_path(@passenger)
+        expect(page).to have_current_path edit_passenger_path(passenger)
       end
     end
 
-    context 'after becoming active' do
-      it 'does not allow editing' do
-        @passenger = create(:temporary_passenger, :with_note)
-        login_as(@passenger)
-        visit edit_passenger_path(@passenger)
-        expect(page).to have_no_field 'Address'
-        expect(page).to have_current_path passenger_path(@passenger)
-        expect(page).to have_no_current_path edit_passenger_path(@passenger)
-        expect(page).to have_text 'To edit your profile, please call'
+    context 'when now active' do
+      let(:passenger) { create(:temporary_passenger, :with_note) }
+
+      it 'redirects to the show page' do
+        expect(page).to have_current_path passenger_path(passenger)
+      end
+
+      it 'redirects to the show page for registering' do
         visit register_passengers_path
+        expect(page).to have_current_path passenger_path(passenger)
+      end
```

Comment:
I don't find the name of this particularly intuitive... it took me a little while to figure out what is going on here. 

maybe this?
```suggestion
      it 'redirects to the show page instead of registering' do
        visit register_passengers_path
        expect(page).to have_current_path passenger_path(passenger)
      end
```